File: src/KeyControl.cpp
#include <allegro.h>

#include "KeyControl.h"


KeyControl::KeyControl(int aKey) {
  keyCode = aKey;
  released = true;
  timer = 0;
  delay = 0; //GAME_TICKER / 4;
}


bool KeyControl::isPressed() {
  
  bool ret = false;
  
  if (delay == 0 && key[keyCode]) {
    return true;
  }
  
  if (key[keyCode]) {
    if (released) {
      
      ret = true;
      timer = 0;
      
    } else {
      timer++;
    }
    
    released = false;
    
    if (timer >= delay) {
      released = true;
    }
    
  } else {
    released = true;
  }
  
  return ret;
  
}


void KeyControl::setDelay(bool theDelay) {
  delay = theDelay;

}

File: src/KeyControl.m
#import "KeyControl.h"


@implementation KeyControl


- initWithKey: (int) aKey {
  self = [super init];
  if (self) {
    keyCode = aKey;
    released = YES;
    timer = 0;
    delay = 0; //GAME_TICKER / 4;
  }
  return self;
}


- (BOOL) isPressed {
  
  BOOL ret = NO;
  
  if (delay == 0 && key[keyCode]) {
    return YES;
  }
  
  if (key[keyCode]) {
    if (released) {
      
      ret = YES;
      timer = 0;
      
    } else {
      timer++;
    }
    
    released = NO;
    
    if (timer >= delay) {
      released = YES;
    }
    
  } else {
    released = YES;
  }
  
  return ret;
  
}


- setDelay: (BOOL) theDelay {
  delay = theDelay;
  return self;
}


@end

